### PR TITLE
prevent updates to host params that should not be changed

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
prevent server overwriting immutable host parameters

## Provide link to Netmaker PR if required
https://github.com/gravitl/netmaker/pull/2335

## Provide testing steps
NET-151
## Checklist before requesting a review
- [ ] My changes affect only 10 files or less.
- [ ] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netclient & Netmaker are awesome.

### CODE REVIEW

**Comment by abhishek9686** [comment]
is there a bug encountered that we are fixing with this PR?
Because the server is the source of truth, why would it be a problem with the data coming from the server?

**Comment by mattkasun** [comment]
> is there a bug encountered that we are fixing with this PR? Because the server is the source of truth, why would it be a problem with the data coming from the server?

version not updating in ui; new firewall not updating.

**Review by abhishek9686 — APPROVED** [review]

**Review by Aceix — APPROVED** [review]

File: functions/mqhandlers.go [hunk 1]
```diff
@@ -351,6 +351,15 @@ func updateHostConfig(host *models.Host) (resetInterface, restart bool) {
...
 		resetInterface = true
 	}
+	// do not update fields that should not be changed by server
+	host.OS = hostCfg.OS
+	host.FirewallInUse = hostCfg.FirewallInUse
+	host.DaemonInstalled = hostCfg.DaemonInstalled
+	host.ID = hostCfg.ID
+	host.Version = hostCfg.Version
+	host.MacAddress = hostCfg.MacAddress
+	host.PublicKey = hostCfg.PublicKey
```
Comment (afeiszli) [inline functions/mqhandlers.go:361]: Does the public key refresh still work with this? If not tested, should be a part of the testing steps just to confirm.

Comment (mattkasun) [reply functions/mqhandlers.go:361]: tested host key update;  confirmed feature still functions

**Review by theguy951357 — APPROVED** [review]
pr steps pass
critical list passes

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**